test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Telemetry/TelemetryOnceEmitterTests.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Moq;
+using NuGet.Common;
+using NuGet.VisualStudio;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    public class TelemetryOnceEmitterTests
+    {
+        private readonly ConcurrentQueue<TelemetryEvent> _telemetryEvents;
+
+        public TelemetryOnceEmitterTests()
+        {
+            var telemetrySession = new Mock<ITelemetrySession>();
+            _telemetryEvents = new ConcurrentQueue<TelemetryEvent>();
+            telemetrySession
+                .Setup(x => x.PostEvent(It.IsAny<TelemetryEvent>()))
+                .Callback<TelemetryEvent>(x => _telemetryEvents.Enqueue(x));
+            TelemetryActivity.NuGetTelemetryService = new NuGetVSTelemetryService(telemetrySession.Object);
+        }
+
+        [Fact]
+        public void EmitIfNeeded_HappyPath_EmitsTelemetryOnce()
```

Comment:
Can you replace "HappyPath" with a descriptive name indicating what makes this path happy? Specifically, what scenario is being tested. Suggestions: EmitIfNeeded_WhenCalledMultipleTimes_EmitsTelemetryOnce

Reply:
Replaced with "EmitIfNeeded_TwoCalls_EmitsTelemetryOnce"